geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -482,5 +481,43 @@ class MapListFunctionFactory extends TransformerFunctionFactory {
         }.toMap.asJava
     }
   }
-
+}
+class CastFunctionFactory extends TransformerFunctionFactory {
+  override def functions = Seq(string2double,string2int,string2float,string2long,string2boolean)
+
+  val string2double = TransformerFn("string2double") {
```

Comment:
We decided to make the conversions functions, so that you have to be explicit about the default value. It was a lot easier to have a function with 2 args than try to change the regex parsers to add a default to the :: syntax. We left the original conversions the same, so they can be used in the case where there is no appropriate default and you want the record to fail.
We could obviously do something else, but that was our thinking.